src/lib.rs

```diff
@@ -5,6 +5,7 @@
 //! input in a cross-platform way.
 
 #![warn(clippy::doc_markdown)]
+#![cfg_attr(docsrs, feature(doc_auto_cfg))]
```

Comment:
What does this line do?

Reply:
The `docsrs` cfg is set by docs.rs when the docs are being built by them. They build using the nightly compiler, so we can use nightly features. [Reference](https://docs.rs/about/builds).

`doc_auto_cfg` is a useful nightly feature, which makes `cfg(...)` attributes show up in the documentation. It looks something like:

<img width="456" alt="screenshot of docs.rs" src="https://github.com/user-attachments/assets/983a281c-d7cf-4e0d-91e1-03ffe8c34aa4" />

See also [the documentation for the feature](https://doc.rust-lang.org/rustdoc/unstable-features.html#doc_auto_cfg-automatically-generate-doccfg).

Reply:
Ah, the screenshot explains  it better than the documentation text.